src/components/input/index.tsx

```diff
@@ -99,15 +101,18 @@ const Input = forwardRef<HTMLInputElement, Props>(
       visible: !!value,
       disabled,
       onClear: () => {
-        onChange({ target: { name, value: "", cleared: true } })
+        const clearEvent = { target: { name, value: "", cleared: true } }
+        onChange(clearEvent)
+        setInputEvent(clearEvent)
       },
     }
+    const [inputValue, setInputEvent] = useDebounce(onChange, debounce)
 
     const renderInput = (hasError) => (
       <InputField
         ref={ref}
         name={name}
-        value={value || ""}
+        value={inputValue}
```

Comment:
I interpret that `inputValue` would be the debounced one. is that really what we want as the value of the input field?

Reply:
No, actually `inputValue` is the immidiate value, the debonced value is passed to the callback function (useDebounce.tsx)